src/v/serde/serde.h

```diff
@@ -634,6 +642,10 @@ void read_nested(iobuf_parser& in, T& t, std::size_t const bytes_left_limit) {
           read_nested<serde_size_t>(in, bytes_left_limit));
         in.consume_to(str.size(), str.begin());
         t = str;
+    } else if constexpr (std::is_same_v<Type, uuid_t>) {
+        for (auto i = 0U; i < uuid_t::length; ++i) {
+            t.uuid.data[i] = read_nested<uint8_t>(in, bytes_left_limit);
```

Comment:
seems like you could do one 16-byte read, right?

Reply:
Done